src/Native/Unix/System.Security.Cryptography.Native/pal_ssl.c

```diff
@@ -676,3 +676,16 @@ int32_t CryptoNative_SslSetTlsExtHostName(SSL* ssl, uint8_t* name)
     return (int32_t)SSL_set_tlsext_host_name(ssl, name);
 }
 
+int32_t CryptoNative_SslGetCurrentCipherId(SSL* ssl, int32_t* cipherId)
+{
+    const SSL_CIPHER* cipher = SSL_get_current_cipher(ssl);
+    if (!cipher)
+    {
+        *cipherId = -1;
+        return 0;
+    }
+
+    *cipherId = SSL_CIPHER_get_id(cipher) & 0xFFFF;
```

Comment:
Why is the `& 0xFFFF` necessary?

Reply:
SSL_CIPHER_get_id returns the OpenSSL 4-byte identifier, which has the IANA value as the low word and some OpenSSL-specific stuff as the high word.

I agree that it's worth a comment :smile:.

Reply:
added comment